engine/src/socket_resolve.cpp

```diff
@@ -434,7 +434,7 @@ bool MCS_name_to_sockaddr(MCStringRef p_name_in, struct sockaddr_in *r_addr, MCH
     
 	memset((char *)r_addr, 0, sizeof(struct sockaddr_in));
 	r_addr->sin_family = AF_INET;
-	r_addr->sin_port = MCSwapInt16HostToNetwork(port);
+	r_addr->sin_port = MCSwapInt16HostToNetwork(uint16_t(port));
```

Comment:
Port needs to be checked for valid range here too (well, actually, probably above near the `MCNumberFetchAsInteger()` call).  Should handle an out-of-range port in the same way as an unrecognised named port.